packages/neos-ui/src/Containers/RightSideBar/Inspector/index.js

```diff
@@ -22,15 +22,23 @@ import style from './style.css';
 @connect((state, {nodeTypesRegistry, validatorRegistry}) => {
     const isApplyDisabledSelector = selectors.UI.Inspector.makeIsApplyDisabledSelector(nodeTypesRegistry, validatorRegistry);
 
-    return state => ({
-        focusedNode: selectors.CR.Nodes.focusedSelector(state),
-        node: selectors.CR.Nodes.focusedSelector(state),
-        isApplyDisabled: isApplyDisabledSelector(state),
-        isDiscardDisabled: selectors.UI.Inspector.isDiscardDisabledSelector(state)
-    });
+    return state => {
+        const isDirty = selectors.UI.Inspector.isDirty(state);
+        const shouldPromptToHandleUnappliedChanges = selectors.UI.Inspector.shouldPromptToHandleUnappliedChanges(state);
+        const shouldShowUnappliedChangesOverlay = isDirty && !shouldPromptToHandleUnappliedChanges;
```

Comment:
Just thinking, wouldn't it be too distracting if the editor is shown this dialog immediately once he starts typing? Maybe it should only appear on unfocus from any of the inspector editors? But yeah, tricky to implement I guess...

Reply:
The dialog isn't shown immediately, but the transparent overlay that blocks everything but the inspector (and secondary inspector). Once the user clicks on that blocking overlay, the dialog pops up.

Reply:
Sorry, I didn't mean the dialog, I meant the overlay. I find it to be a bit distracting, and would prefer to see it only on unfocus, but I guess it's ok the way it is.

Reply:
How is it distracting? It isn't visible :)

Reply:
Lol, by looking at the code I thought it is. So it's there only to protect from clicking outside    of inspector?

Reply:
Lol, by looking at the code I thought it is. So it's there only to protect from clicking outside    of inspector?